src/test/java/io/cdap/plugin/gcp/dataplex/source/DataplexBatchSourceTest.java

```diff
@@ -0,0 +1,119 @@
+package io.cdap.plugin.gcp.dataplex.source;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import io.cdap.cdap.etl.mock.validation.MockFailureCollector;
+import io.cdap.plugin.gcp.common.GCPConnectorConfig;
+import io.cdap.plugin.gcp.dataplex.common.config.DataplexBaseConfig;
+import io.cdap.plugin.gcp.dataplex.source.config.DataplexBatchSourceConfig;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.internal.util.reflection.FieldSetter;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.powermock.api.mockito.PowerMockito;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+/**
+ * Tests for DataplexBatchSource
+ */
+@RunWith(MockitoJUnitRunner.class)
+//@PrepareForTest({DataplexBatchSource.class, DataplexBatchSourceConfig.class, BigQueryUtil.class})
+public class DataplexBatchSourceTest {
+  @Mock
+  private GCPConnectorConfig connection;
+
+  @Test
+  public void tryGetProjectTest() {
+    DataplexBatchSourceConfig config = spy(new DataplexBatchSourceConfig());
+    when(config.tryGetProject()).thenReturn(null);
+    assertNull(config.tryGetProject());
+    verify(config, times(1)).tryGetProject();
+  }
+
+  @Test
+  public void validateBigQueryDatasetTest() throws IOException {
+    DataplexBatchSourceConfig dataplexBatchSourceConfig = new DataplexBatchSourceConfig();
+    MockFailureCollector mockFailureCollector = new MockFailureCollector();
+   // BigQueryUtil bigQueryUtil = spy(new BigQueryUtil());
+/*    Table table = mock(Table.class);
+    when(BigQueryUtil.getBigQueryTable(anyString(), anyString(), anyString(), anyString(), anyBoolean(), any(
+      FailureCollector.class))).thenReturn(table);*/
+    dataplexBatchSourceConfig.validateBigQueryDataset(mockFailureCollector, "projectId", "dataset", "table");
+    assertEquals(0, mockFailureCollector.getValidationFailures().size());
+  }
+
+  @Test
+  public void validateServiceAccount() throws Exception {
+    MockFailureCollector mockFailureCollector = new MockFailureCollector();
+    DataplexBatchSourceConfig config = spy(new DataplexBatchSourceConfig());
+    GCPConnectorConfig gcpConnectorConfig = spy(new GCPConnectorConfig("", "", "", ""));
+    FieldSetter.setField(config, DataplexBaseConfig.class.getDeclaredField("connection"),
```

Comment:
instead of `FieldSetter` can we use builder with Private Constructor in all the tests. This makes the tests brittle if we modified the field name in future.

Reply:
done